src/hoc/appInitializer.js

```diff
@@ -0,0 +1,27 @@
+import {useEffect, useState} from "react";
+import {useDispatch, useSelector} from "react-redux";
+import {loginSuccess} from "../store/auth/authActions";
+import LoadingScreen from "../components/Layout/LoadingScreen";
+
+const appInitializer = WrappedComponent => props => {
+    const [loading, setLoading] = useState(false);
+    const {user} = useSelector(state => state.authReducer);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        setLoading(true);
```

Comment:
why not set loading initially to true? then you don't need to set it in useEffect